[CRASH] App crashes when using Biometric Api

Castle integration causes a crash due to the fact that `DeviceCredentialHandlerActivity` sets its title to null.


Reference:
https://android.googlesource.com/platform/frameworks/support/+/refs/heads/androidx-recyclerview-release/biometric/src/main/java/androidx/biometric/DeviceCredentialHandlerActivity.java#54

**Device**: Samsung S7 Edge (SM-G935F)
**Android**: 8.0.0

Stacktrace:
```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.stash.stashinvest, PID: 10195
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.stash.stashinvest/androidx.biometric.DeviceCredentialHandlerActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'java.lang.String java.lang.CharSequence.toString()' on a null object reference
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2957)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3032)
        at android.app.ActivityThread.-wrap11(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1696)
        at android.os.Handler.dispatchMessage(Handler.java:105)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6944)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
     Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'java.lang.String java.lang.CharSequence.toString()' on a null object reference
        at io.castle.android.api.model.ScreenEvent.<init>(ScreenEvent.java:30)
        at io.castle.android.Castle.screen(Castle.java:276)
        at io.castle.android.CastleActivityLifecycleCallbacks.onActivityStarted(CastleActivityLifecycleCallbacks.java:18)
        at android.app.Application.dispatchActivityStarted(Application.java:231)
        at android.app.Activity.onStart(Activity.java:1296)
        at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:536)
        at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:201)
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1340)
        at android.app.Activity.performStart(Activity.java:7200)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2920)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3032) 
        at android.app.ActivityThread.-wrap11(Unknown Source:0) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1696) 
        at android.os.Handler.dispatchMessage(Handler.java:105) 
        at android.os.Looper.loop(Looper.java:164) 
        at android.app.ActivityThread.main(ActivityThread.java:6944) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374) 
```

--- a/castle/src/main/java/io/castle/android/api/model/ScreenEvent.java
+++ b/castle/src/main/java/io/castle/android/api/model/ScreenEvent.java
@@ -27,6 +27,6 @@ public class ScreenEvent extends Event {
      * @param activity Activity
      */
     public ScreenEvent(Activity activity) {
-        this(activity.getTitle().toString());
+        this(activity.getTitle() != null ? activity.getTitle().toString() : activity.getClass().getSimpleName());
     }
 }

--- a/castle/src/test/java/io/castle/android/CastleTest.java
+++ b/castle/src/test/java/io/castle/android/CastleTest.java
@@ -46,6 +46,8 @@ public class CastleTest {
     public void setup() {
         application = rule.getActivity().getApplication();
 
+        rule.getActivity().setTitle("TestActivityTitle");
+
         ArrayList<String> baseUrlWhiteList = new ArrayList<>();
         baseUrlWhiteList.add("https://google.com/");
 
@@ -131,6 +133,17 @@ public class CastleTest {
 
         ScreenEvent screenEvent = new ScreenEvent("Main");
         Assert.assertEquals(screenEvent.getEvent(), "Main");
+        Assert.assertEquals(screenEvent.getType(), Event.EVENT_TYPE_SCREEN);
+
+        screenEvent = new ScreenEvent(rule.getActivity());
+        Assert.assertEquals(screenEvent.getEvent(), "TestActivityTitle");
+        Assert.assertEquals(screenEvent.getType(), Event.EVENT_TYPE_SCREEN);
+
+        // Test null activity title
+        rule.getActivity().setTitle(null);
+
+        screenEvent = new ScreenEvent(rule.getActivity());
+        Assert.assertEquals(screenEvent.getEvent(), "TestActivity");
         Assert.assertEquals(screenEvent.getType(), Event.EVENT_TYPE_SCREEN);
 
         count = Castle.queueSize();

